src/main/kotlin/com/craftschemers/hub/managers/HubCommandManager.kt

```diff
@@ -0,0 +1,35 @@
+package com.craftschemers.hub.managers
+
+import com.craftschemers.hub.commands.ICommand
+import com.craftschemers.hub.commands.JoinCommand
+import com.craftschemers.hub.minigame.Minigame
+import org.bukkit.ChatColor
+import org.bukkit.command.Command
+import org.bukkit.command.CommandExecutor
+import org.bukkit.command.CommandSender
+
+class HubCommandManager : CommandExecutor {
+    companion object {
+        private val commands = mutableMapOf<String, ICommand>()
+        init {
+            fun registerCommand(command: ICommand) {
+                commands[command.name] = command
+            }
+            registerCommand(JoinCommand())
+            // ...
+        }
+    }
+
+    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
+        if (args.isEmpty()) {
+            sender.sendMessage("Welcome to the ${ChatColor.GOLD}Hub Plugin!")
+            return true
+        }
+        if (!commands.containsKey(args[0])) {
```

Comment:
Nit, could be `!in`
```suggestion
        if (args[0] !in commands) {
```

Reply:
You could also take advantage of the returned value being `null` if it doesn't have it which might make the code a little more terse.